splinepy/bezier.py

```diff
@@ -0,0 +1,289 @@
+
+    def __add__(self, summand):
+      """
+      Calculates the spline that formes the sum of the summand and the current
+      spline (function argument)
+
+      The resulting spline fulfils the equation
+        self(t) + summand(t) = result(t)
+
+      Parameters
+      ----------
+      summand :  spline with same parametric and physical dimension
+
+      Returns
+      -------
+      spline : New spline that describes sum
+      """
+      if isinstance(summand, type(self)):
+          if summand.para_dim == self.para_dim and \
+            summand.dim == self.dim:
+              resulting_c_spline = \
+                  self._c_spline.add_spline(summand._c_spline)
+              # Copy the c spline to the python object
+              result = type(self)()
+              result._c_spline = resulting_c_spline
+              result._update_p()
+              return result
+          else :
+              raise TypeError("Dimension Mismatch.")          
+      else :
+          raise TypeError("Sum must be formed with Bezier Splines")
+
+
+    def compose(self, inner_function):
+      """
+      Calculates the spline that formes the composition of the inner function
+      spline (function argument), using the caller spline as the outer (or 
+      deformation function).
+
+      The resulting spline fulfils the equation
+        spline_self(inner_function(t)) = result(t)
+
+      Parameters
+      ----------
+      inner_function :  spline with parametric dimension <= 3
+
+      Returns
+      -------
+      spline : New spline that describes composition
+      """
+      if isinstance(inner_function, type(self)):
+          if inner_function.para_dim == 1:
+              resulting_c_spline = \
+                  self._c_spline.compose_line(inner_function._c_spline)
+          elif inner_function.para_dim == 2:
+              resulting_c_spline = \
+                  self._c_spline.compose_surface(inner_function._c_spline)
+          elif inner_function.para_dim == 3:
+              resulting_c_spline = \
+                  self._c_spline.compose_volume(inner_function._c_spline)
+          else :
+              raise TypeError("Compositions with high"
+                      " parametric dimensions not supported.")          
+          # Copy the c spline to the python object
+          result = type(self)()
+          result._c_spline = resulting_c_spline
+          result._update_p()
+          return result
+      else :
+          raise TypeError("Composisiton must be formed with Bezier Splines")
+      
+
+    def copy(self):
+        """
+        """
+        pass
+
+    def todict(self):
+        """
+        Returns copy of degrees, control_points in dict.
+
+        Parameters
+        -----------
+        None
+
+        Returns
+        --------
+        dict_spline: dict
+          Keys are {degrees, control_points}.
+        """
+        return dict(
+            degrees=copy.deepcopy(self.degrees).tolist(),
```

Comment:
`tolist()` should be optional through a param `tolist`.
I am wondering if we could avoid implementing `todict` for each class by defining additional property that returns names of spline property.
Can we discuss this?